core/rio/hdt/src/main/java/org/eclipse/rdf4j/rio/hdt/part/HDTDictionarySectionPFC.java

```diff
@@ -0,0 +1,130 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+package org.eclipse.rdf4j.rio.hdt.part;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.zip.CheckedInputStream;
+
+import org.eclipse.rdf4j.rio.hdt.util.CRC32;
+import org.eclipse.rdf4j.rio.hdt.util.CRC8;
+import org.eclipse.rdf4j.rio.hdt.util.VByte;
+
+/**
+ * HDT DictionarySection Plain Front Coding.
+ * 
+ * This part starts with a byte indicating the type of the dictionary section, followed by the VByte-encoded number of
+ * strings, the VByte-encoded buffer size and the VByte-encoded buffer length.
+ *
+ * Then the 8-bit CRC.
+ * 
+ * Followed by an array and one or more buffers, and the 32-bit CRC calculated over the index and the buffers.
+ * 
+ * Structure:
+ * 
+ * <pre>
+ * +------+-----------+---------+-------+------+-------+--------+...+-------+
+ * | type | nrstrings | bufsize | array | CRC8 | index | buffer |...| CRC32 | 
+ * +------+-----------+---------+-------+------+-------+--------+...+-------+
+ * </pre>
+ * 
+ * Each buffer starts with a full string, followed by a maximum of <code>bufsize</code> - 1 pair of a VByte-encoded
+ * number of characters this string has in common with the _previous_ string, and the (different) suffix.
+ * 
+ * E.g. <code>abcdef 2 gh 3 ij</code> will result in <code>abcde, abgh, abgij</code>.
+ * 
+ * Buffer structure:
+ * 
+ * <pre>
+ * +--------+--------+--------+...+--------+--------+
+ * | string | common | suffix |...| common | suffix |
+ * +--------+--------+--------+...+--------+--------+
+ * </pre>
+ * 
+ * @author Bart Hanssens
+ */
+public class HDTDictionarySectionPFC extends HDTDictionarySection {
+	private ArrayList<byte[]> arr;
+	private int nrstrings;
+	private long buflen;
+	private int bufsize;
+
+	@Override
+	public int size() {
+		return nrstrings;
+	}
+
+	@Override
+	public byte[] get(int i) {
+		return arr.get(i - 1);
+	}
+
+	@Override
+	public void parse(InputStream is) throws IOException {
+		CRC8 crc8 = new CRC8();
+		crc8.update((byte) HDTDictionarySection.Type.FRONT.getValue());
+		CheckedInputStream cis = new CheckedInputStream(is, crc8);
+
+		long val = VByte.decode(cis);
+		if (nrstrings > Integer.MAX_VALUE) {
+			throw new UnsupportedOperationException("Maximum number of strings in dictionary exceeded: " + val);
+		}
+		nrstrings = (int) val;
+
+		buflen = VByte.decode(cis);
+
+		val = VByte.decode(cis);
+		if (val > Integer.MAX_VALUE) {
+			throw new UnsupportedOperationException("Maximum number of bufsize in dictionary exceeded: " + val);
+		}
+		bufsize = (int) val;
+
+		checkCRC(cis, is, 1);
+
+		HDTArray ha = HDTArrayFactory.parse(is);
+		ha.parse(is);
+
+		cis = new CheckedInputStream(is, new CRC32());
+		arr = new ArrayList(nrstrings);
```

Comment:
The generics type or diamond-syntax is missing here. Can use `ArrayList<>(nrstrings)` rather than actually putting the type in.

Reply:
good catch